components/formats-bsd/src/loci/formats/in/DicomTag.java

```diff
@@ -0,0 +1,516 @@
+    }
+    else if (elementLength == 0) {
+      in.seek(in.getFilePointer() - 4);
+      DicomVR v = DicomVR.get(in.readShort() & 0xffff);
+      if (v == UT) {
+        in.skipBytes(2);
+        elementLength = in.readInt();
+      }
+      else in.skipBytes(2);
+    }
+
+    // HACK - needed to read some GE files
+    // The element length must be even!
+    if (!oddLocations && (elementLength % 2) == 1) elementLength++;
+
+    return tag;
+  }
+
+  /**
+   * Get the length of the next value in the given input stream.
+   */
+  private int getLength(RandomAccessInputStream in) throws IOException {
+    byte[] b = new byte[4];
+    in.read(b);
+
+    // We cannot know whether the VR is implicit or explicit
+    // without the full DICOM Data Dictionary for public and
+    // private groups.
+
+    // We will assume the VR is explicit if the two bytes
+    // match the known codes. It is possible that these two
+    // bytes are part of a 32-bit length for an implicit VR.
+
+    // see http://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_7.1.2
+
+    vr = DicomVR.get(((b[0] & 0xff) << 8) | (b[1] & 0xff));
+
+    if (vr == null) {
+      vr = IMPLICIT;
+      int len = DataTools.bytesToInt(b, in.isLittleEndian());
+      if (len + in.getFilePointer() > in.length() || len < 0) {
+        len = DataTools.bytesToInt(b, 2, 2, in.isLittleEndian());
+        len &= 0xffff;
+      }
+      return len;
+    }
+
+    switch (vr) {
+      case OB:
+      case OW:
+      case SQ:
+      case UN:
+      case UT:
+      case UC:
+        // Explicit VR with 32-bit length if other two bytes are zero
+        if ((b[2] == 0) || (b[3] == 0)) {
+          return in.readInt();
+        }
+        vr = IMPLICIT;
+        return DataTools.bytesToInt(b, in.isLittleEndian());
+      case AE:
+      case AS:
+      case AT:
+      case CS:
+      case DA:
+      case DS:
+      case DT:
+      case FD:
+      case FL:
+      case IS:
+      case LO:
+      case LT:
+      case PN:
+      case SH:
+      case SL:
+      case SS:
+      case ST:
+      case TM:
+      case UI:
+      case UL:
+      case US:
+      case QQ:
+        // Explicit VR with 16-bit length
+        if (attribute == LUT_DATA) {
+          return DataTools.bytesToInt(b, 2, 2, in.isLittleEndian());
+        }
+        int n1 = DataTools.bytesToShort(b, 2, 2, in.isLittleEndian());
+        int n2 = DataTools.bytesToShort(b, 2, 2, !in.isLittleEndian());
+        n1 &= 0xffff;
+        n2 &= 0xffff;
+        if (n1 < 0 || n1 + in.getFilePointer() > in.length()) return n2;
+        if (n2 < 0 || n2 + in.getFilePointer() > in.length()) return n1;
+        return n1;
+      case RESERVED:
+        int len = DataTools.bytesToInt(b, 0, 4, in.isLittleEndian());
+        if (len == 0xffffffff) {
+          return len;
+        }
+        else {
+          vr = IMPLICIT;
+          return 8;
+        }
+      default:
+        throw new IllegalArgumentException(vr.toString());
```

Comment:
As a result of the DICOM reader priority increase in `readers.txt`, this exception is now being thrown across several file formats during the `isThisType` check.

See e.g. [bd-pathway](https://merge-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-test-folder/153391/console), [deltavision](https://merge-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-test-folder/153394/console) for examples of affected filesets.

Reply:
I'd expect f75b8dc, 84e1a80, and fc13799 to resolve unexpected test failures, but will check the results tomorrow now that builds are running again.